tabix.c

```diff
@@ -138,89 +167,120 @@ static char **parse_regions(char *regions_fname, char **argv, int argc, int *nre
     if ( !ireg )
     {
         if ( argc )
+        {
             regs = (char**) malloc(sizeof(char*)*argc);
+            if (!regs) error_errno(NULL);
+        }
         else
         {
             regs = (char**) malloc(sizeof(char*));
+            if (!regs) error_errno(NULL);
             regs[0] = strdup(".");
+            if (!regs[0]) error_errno(NULL);
             *nregs = 1;
         }
     }
 
-    for (iseq=0; iseq<argc; iseq++) regs[ireg++] = strdup(argv[iseq]);
+    for (iseq=0; iseq<argc; iseq++, ireg++) {
+        regs[ireg] = strdup(argv[iseq]);
+        if (!regs[ireg]) error_errno(NULL);
+    }
     return regs;
 }
 static int query_regions(args_t *args, char *fname, char **regs, int nregs, int download)
 {
     int i;
     htsFile *fp = hts_open(fname,"r");
-    if ( !fp ) error("Could not read %s\n", fname);
+    if ( !fp ) error_errno("Could not open \"%s\"", fname);
     enum htsExactFormat format = hts_get_format(fp)->format;
 
     regidx_t *reg_idx = NULL;
     if ( args->targets_fname )
     {
         reg_idx = regidx_init(args->targets_fname, NULL, NULL, 0, NULL);
-        if ( !reg_idx ) error("Could not read %s\n", args->targets_fname);
+        if (!reg_idx) error_errno("Could not read \"%s\"", args->targets_fname);
```

Comment:
There are a number of reasons for which `regidx_init` can fail. Perhaps the error message could be more indicative (e.g. "Failed to build region information from file: \\"%s\\"") so as to not make the user think there is just a problem with reading the file.

Reply:
Text has been amended.